### PR TITLE
fix: #20002 optimize head request on serializeDataProvider

### PR DESCRIPTION
**Pull request by xicond**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #20002

### CODE REVIEW

**Comment by xicond** [comment]
@bizley 

I tried my best to add unit test which not covered before, seems sentry still not satisfy and not detect the coverage
Also the test only for previous state, not testing #20002 kind of bug, 

to test #20002, I'd need to include Mockery library $arrayDataProviderMocked->shouldReceive('getModels')->never()
But I dont see its usage before, so I think it's not necessary.

**Comment by bizley** [comment]
I'm not sure if adding it in the same test is not interfering with it. Try to exclude it to a new one. Also, Mockery is not needed, PHPUnit is enough.

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,9 @@ Yii Framework 2 Change Log
 
 - Bug #13920: Fixed erroneous validation for specific cases (tim-fischer-maschinensucher)
 - Bug #19927: Fixed `console\controllers\MessageController` when saving translations to database: fixed FK error when adding new string and language at the same time, checking/regenerating all missing messages and dropping messages for unused languages (atrandafir)
+- Bug #20002: fixed superfluous query on HEAD request in serializer (xicond)
```
Comment (samdark) [inline framework/CHANGELOG.md:9]:
```suggestion
- Bug #20002: Fixed superfluous query on HEAD request in serializer (xicond)
```

**Review by samdark — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

**Comment by xicond** [comment]
I separated the test, codeconv back to be not satisfied 
and last time nodejs test [17871509197](https://github.com/yiisoft/yii2/actions/runs/6578233064/job/17871509197?pr=20007) failed, now it's success, seems some tests are unstable

File: tests/framework/rest/SerializerTest.php [hunk 2]
```diff
@@ -400,6 +402,11 @@ public function dataProviderSerializeDataProvider()
         ];
     }
 
+    public function dataProviderHeadSerializeDataProvider()
```
Comment (bizley) [inline tests/framework/rest/SerializerTest.php:405]: This is unnecessary. Just use the same provider method in your test.

**Review by bizley — APPROVED** [review]

**Comment by bizley** [comment]
Cheers

**Pull request merged**